Implemented store infrastructure for cockroachDb

vicpatel opened the pull request:
- Implemented store methods for cockroachDB
- Updated DDL generator for cockroachDB to create/update tables and indexes
- Workaround for a LIKE operator in cockroachDB
- Added sqlFunctionHandler and fieldToColumnTypeCaster handle for cockroachDB

Please take a look.

<!--
Checklist:

1. API changes have been discussed,
2. Source code must be formatted (`go fmt`),
3. Codegen shouldn't produce modified files,
4. Builds pass,
5. Tests pass,
6. Linked to relevant issues

When you are writing pull request title and describing changes, follow
commit message format in the CONTRIBUTING.md in the codebase root.

-->

darh commented on store/cockroach/sql_typecasters.go:
```diff
@@ -0,0 +1,16 @@
+//   * field cast tpl: fmt template to get query column
+//   * type cast tpl: fmt template to cast the compared to value
+func fieldToColumnTypeCaster(field rdbms.ModuleFieldTypeDetector, ident string) (string, string, string, error) {
+	return "", "", "", nil
```
why is this empty? does none of the current implementations (my/pg/lite) fit?

vicpatel replied:
Added it, It does!

darh commented on store/cockroach/sql_functions.go:
```diff
@@ -0,0 +1,10 @@
+)
+
+func sqlFunctionHandler(f ql.Function) (ql.ASTNode, error) {
+	// @todo ^^
```
why is this empty? does none of the current implementations (my/pg/lite) fit?

vicpatel replied:
Added it, yes it does!

darh reviewed (commented):
Great! Getting closer to working version.

darh commented on store/cockroach/cockroach.go:
```diff
@@ -0,0 +1,113 @@
+//
+// @note we could enable this once cockroachDB overcome above limitations :)
+func init() {
+	//store.Register(Connect, storeTypeName, storeTypeWithDebug)
```
How come this is commented out?
Can we leave it enabled? Since the cockroachdb package (this) itself is not imported it does not matter what we have in the init fn()...

vicpatel replied:
Yes, I Will add it back!

darh commented on store/cockroach/cockroach.go:
```diff
@@ -0,0 +1,113 @@
+
+const (
+	storeTypeName = "cockroachdb"
+	// storeTypeWithDebug = "cockroachdb+debug"
```
same as the comment below for the init() fn.

vicpatel replied:
Sure

vicpatel committed:
Implements store infrastructure for cockroachDB

Provides cockroachdb support but due to some limitation from cockroachDB, we could not implement it further.

Development so far,
- Implements store methods
- Updates DDL generator to create/update tables and indexes
- Provides sqlFunctionHandler, fieldToColumnTypeCaster handle, and workaround for LIKE operator
